core/variables.js

```diff
@@ -289,9 +289,19 @@ Blockly.Variables.delete = function(name, workspace) {
   if (variableIndex != -1) {
     var uses = Blockly.Variables.getUses(name, workspace);
     if (uses.length > 1) {
+      for (var i = 0, block; block = uses[i]; i++) {
```

Comment:
General, unrelated to your code, question: do you know where we validate/sanitize the procedure name the user gives us?